### PR TITLE
Adds rationale for api.ValueType encoding

### PR DESCRIPTION
**Pull request by codefromthecrypt**
We made some decisions around encoding last year, but we didn't detail why. This leads to questions.

e.g. https://gophers.slack.com/archives/C040AKTNTE0/p1668883638982119

### CODE REVIEW

File: RATIONALE.md [hunk 1]
```diff
@@ -112,6 +115,65 @@ Here are some examples:
 
 Besides security, this practice prevents other bugs and allows centralization of validation logic such as decoding Wasm.
 
+## API Design
+
+### Why are Go function signature parameters and results uint64?
+
+Go-defined function signatures include parameters and results expressed as a
```
Comment (mathetake) [inline RATIONALE.md:122]: Isn't this only for the Call API? Host functions can naturally have any types other than u64

**Review by mathetake — APPROVED** [review]

File: RATIONALE.md (hunk 1 above)
Comment (codefromthecrypt) [reply RATIONALE.md:122]: good point. I guess I didn't write what I meant.. we have two apis to call functions from go..

I'll try to rewrite a bit as this is about invocation from go

Comment (codefromthecrypt) [reply RATIONALE.md]: PTAL!

**Review by mathetake — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
ps @jerbob92 mentions

> It makes sense but still think it's confusing for developers that you're expecting something with the size of a uint64, and not a literal uint64. 

imho agreed it is confusing, and hopefully the ecosystem will use/make things like [go-plugin](https://github.com/knqyf263/go-plugin) or a future [wit/witx binder](#662) to reduce exposure of end users. meanwhile we can expect some questions, but at least RATIONALE is here for folks in the community to help answer them

**Pull request merged**